$OpenBSD: patch-tools_clang_lib_Driver_ToolChains_Arch_PPC_h,v 1.2 2018/11/19 08:51:03 ajacoutot Exp $

- [PowerPC] Option for secure plt mode
- [PowerPC] Set the default PLT mode on OpenBSD/powerpc to Secure PLT.

Index: tools/clang/lib/Driver/ToolChains/Arch/PPC.h
--- tools/clang/lib/Driver/ToolChains/Arch/PPC.h.orig
+++ tools/clang/lib/Driver/ToolChains/Arch/PPC.h
@@ -29,10 +29,17 @@ enum class FloatABI {
   Hard,
 };
 
+enum class ReadGOTPtrMode {
+  Bss,
+  SecurePlt,
+};
+
 FloatABI getPPCFloatABI(const Driver &D, const llvm::opt::ArgList &Args);
 
 std::string getPPCTargetCPU(const llvm::opt::ArgList &Args);
 const char *getPPCAsmModeForCPU(StringRef Name);
+ReadGOTPtrMode getPPCReadGOTPtrMode(const Driver &D, const llvm::Triple &Triple,
+                                    const llvm::opt::ArgList &Args);
 
 void getPPCTargetFeatures(const Driver &D, const llvm::Triple &Triple,
                           const llvm::opt::ArgList &Args,
